packages/api-explorer/src/components/SideNav/SideNav.tsx

```diff
@@ -111,15 +110,25 @@ export const SideNav: FC<SideNavProps> = ({ headless = false, spec }) => {
     setSearchPattern(value)
   }
 
+  useEffect(() => {
+    if (debouncedPattern && debouncedPattern !== searchParams.get('s')) {
+      searchParams.set('s', debouncedPattern)
+      navigate(location.pathname, { search: searchParams.toString() })
+    } else if (!debouncedPattern && searchParams.get('s')) {
+      searchParams.delete('s')
+      navigate(location.pathname, { search: searchParams.toString() })
+    }
+  }, [debouncedPattern])
```

Comment:
searchParams should probably be a dependency

Reply:
Just to clarify, do you recommend I do this or move searchParams in the effect and add location.search as a dependency as you mentioned in the last comment? If I instantiate searchParams in the effect I wouldn't be able to also add it as a dependency.

Reply:
I would add the dependency but thats because I'm a conformist :D . See previous comment